src/slam_toolbox_sync.cpp

```diff
@@ -69,6 +66,34 @@ void SynchronousSlamToolbox::run()
   }
 }
 
+/*****************************************************************************/
+CallbackReturn
+SynchronousSlamToolbox::on_configure(const rclcpp_lifecycle::State &)
+/*****************************************************************************/
+{
+  processor_type_ = PROCESS;
+  SlamToolbox::on_configure(rclcpp_lifecycle::State());
+  ssClear_ = this->create_service<slam_toolbox::srv::ClearQueue>(
+    "slam_toolbox/clear_queue",
+    std::bind(
+      &SynchronousSlamToolbox::clearQueueCallback, this,
+      std::placeholders::_1, std::placeholders::_2, std::placeholders::_3));
+  threads_.push_back(
+    std::make_unique<boost::thread>(
+      boost::bind(&SynchronousSlamToolbox::run, this)));
+  return CallbackReturn::SUCCESS;
+}
+
+/*****************************************************************************/
+CallbackReturn
+SynchronousSlamToolbox::on_cleanup(const rclcpp_lifecycle::State &)
+/*****************************************************************************/
+{
+  SlamToolbox::on_cleanup(rclcpp_lifecycle::State()); // with run threads cleanup
+  ssClear_.reset();
```

Comment:
This doesn't clear the threads in the vector

Reply:
It is cleared in `on_cleanup`

<https://github.com/SteveMacenski/slam_toolbox/blob/2ea6b342054962171e9278caef9ff3d96b6b8cc4/src/slam_toolbox_common.cpp#L98-L104>

In relation to https://github.com/SteveMacenski/slam_toolbox/pull/561#discussion_r1056317665 , I intend to move this clear process to `on_deactivate` func.